thirdparty/tesseract/CMakeLists.txt

```diff
@@ -6,25 +6,35 @@ include("koreader_thirdparty_common")
 include("koreader_thirdparty_git")
 
 ep_get_source_dir(SOURCE_DIR)
+ep_get_binary_dir(BINARY_DIR)
 
 if ("$ENV{ANDROID}" STREQUAL "1")
-    set(PATCH_CMD "${KO_PATCH} ${CMAKE_CURRENT_SOURCE_DIR}/tesseract-android.patch")
+    set(CXXFLAGS "${CXXFLAGS} -DANDROID")
 endif()
 
+set(CFG_ENV_VAR "PKG_CONFIG=/bin/false CC=\"${CC}\" CXX=\"${CXX}\" CFLAGS=\"${CFLAGS}\" CXXFLAGS=\"${CXXFLAGS}\" LEPTONICA_CFLAGS=\" \" LEPTONICA_LIBS=\" \" LDFLAGS=\"${LDFLAGS}\" LIBS=\"${LIBS}\"")
+set(CFG_OPTS "-q --prefix=${BINARY_DIR} --disable-static --enable-shared --disable-openmp --enable-visibility --host=\"${CHOST}\"")
+set(CFG_CMD sh -c "${CFG_ENV_VAR} ${SOURCE_DIR}/configure ${CFG_OPTS}")
+
+# Add errno for android until tesseract fixes it
+set(PATCH_CMD sh -c "${ISED} 's/#include <unistd\.h>/#include <unistd.h>\\n#include <errno.h>/' ${SOURCE_DIR}/src/api/baseapi.cpp")
+set(PATCH2_CMD "${KO_PATCH} ${CMAKE_CURRENT_SOURCE_DIR}/tesseract-k2pdfopt.patch")
+set(PATCH3_CMD sh -c "${ISED} '/bin_PROGRAMS = tesseract/d' ${SOURCE_DIR}/src/api/Makefile.am")
+
+
 ko_write_gitclone_script(
     GIT_CLONE_SCRIPT_FILENAME
     https://github.com/tesseract-ocr/tesseract.git
-    60176fc5ae5e7f6bdef60c926a4b5ea03de2bfa7
+    4.1
     ${SOURCE_DIR}
 )
 
 include(ExternalProject)
 ExternalProject_Add(
     ${PROJECT_NAME}
     DOWNLOAD_COMMAND ${CMAKE_COMMAND} -P ${GIT_CLONE_SCRIPT_FILENAME}
-    PATCH_COMMAND COMMAND ${PATCH_CMD}
-    CONFIGURE_COMMAND ""
-    # skip build and install, libk2pdfopt will build it
-    BUILD_COMMAND ""
-    INSTALL_COMMAND ""
+    PATCH_COMMAND COMMAND ${PATCH_CMD} COMMAND ${PATCH2_CMD} COMMAND ${PATCH3_CMD} COMMAND sh -c "${ISED} \"s|\\$srcdir\\/configure \\$\\*||g\" autogen.sh" COMMAND ./autogen.sh
+    CONFIGURE_COMMAND ${CFG_CMD}
+    BUILD_COMMAND ${KO_MAKE_RECURSIVE} -j1 # TBA: This is heavy weight C++, -j1 necessary so as to not nuke CircleCI instances
```

Comment:
@Frenzie Is there a good way to make this conditional only for CI instances that are limited by memory (but without snipping parallism overall)? The reason this goes kaboom on CircleCI is ginormous C++ template expansions in Tesseract 4.

Reply:
Don't CI build envs usually set an environment variable to detect 'em? (Or can we ask 'em to do that, somehow?)

Reply:
Yes, there are various environment variables including a fairly self-evident `${CIRCLECI}`, see https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables for the full list.

Reply:
Turns out CMAKE can tell us available memory directly, so I suppose people with old computers can appreciate us not bombing their box too :>

Reply:
> need at least 2.5GB per CPU

Which would include my personal laptop at 4 GB which is perfectly usable as long as e.g. tesseract is only compiled irregularly, so yes, much better. ;-)